src/hotspot/share/oops/methodData.hpp

```diff
@@ -2511,7 +2511,7 @@ class MethodData : public Metadata {
 
   void clean_method_data(bool always_clean);
   void clean_weak_method_links();
-  Mutex* extra_data_lock() { return &_extra_data_lock; }
+  Mutex* extra_data_lock() const { return const_cast<Mutex*>(&_extra_data_lock); }
```

Comment:
Not sure what this const-ness gains us :)

Reply:
It gets us to compile line 1561 above in the print function which is const.

Reply:
Ah, right.